src/core/ext/transport/chttp2/client/secure/secure_channel_create.c

```diff
@@ -276,10 +271,10 @@ static grpc_channel *client_channel_factory_create_channel(
                                               GRPC_CLIENT_CHANNEL, NULL);
   grpc_channel_args_destroy(final_args);
 
-  grpc_resolver *resolver = grpc_resolver_create(target, &f->base);
+  grpc_resolver *resolver = grpc_resolver_create(target);
   if (resolver != NULL) {
-    grpc_client_channel_set_resolver(
-        exec_ctx, grpc_channel_get_channel_stack(channel), resolver);
+    grpc_client_channel_set_resolver_and_client_channel_factory(
```

Comment:
If you'd prefer, I can either leave this method with its original name, or I can provide two different methods, one for each thing we need to set.

As an aside, I've actually been thinking that the client channel factory interface is a bit messy and that we should come up with a way to clean it up.  But I don't yet have any concrete thoughts on how to do that.